jobspec/parse_test.go

```diff
@@ -128,6 +128,20 @@ func TestParse(t *testing.T) {
 									MaxFiles:      10,
 									MaxFileSizeMB: 100,
 								},
+								Artifacts: []*structs.TaskArtifact{
```

Comment:
Can we add a test where we pass invalid keys to Artifact options and have the parser fail?